### DISCUSSION TITLE
hook eme_trash_rsvp_action

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#563] [General]
Hi
I have a snippet that runs on the eme_trash_rsvp_action hook.
When the booker cancels his booking, or an administrator cancels a booking, this code is executed and works very well.
But when the author of an event, or an admin, cancels the event and sends it to the trash, the related bookings are also sent to the trash. But my snippet doesn't work. 
Is this because eme_trash_rsvp_action isn't used in this case?

**Comment by liedekef** [comment]
Well, there was (is) code optimization in place if you chose not to send mails concerning the bookings (and unintentionally, that then doesn't execute the mentioned hook either).
This should fix it:
https://github.com/liedekef/events-made-easy/commit/f0bc2b4407934dcfcc5bcbf80c7a62b424fa10bc

**Reply by f14m07** [reply]
I tested with the new eme_events.php, but it doesn't change anything, whether I answer yes or no to the question:	“Send mails for cancelled bookings too”?

**Reply by liedekef** [reply]
The function eme_trash_booking executes the hook, and the function that trashes the event (eme_trash_events) now calls eme_trash_booking if either the mails need to be sent or the action hook exists).

**Reply by f14m07** [reply]
So I don't understand why it's Ok when I, or the person, delete the booking, and not when I delete the event (with related booking). My snippet is hooked on eme_trash_rsvp_action, call by eme_trash_booking but it doesn't work on this case

**Reply by liedekef** [reply]
I already explained that eme_trash_booking executes the hook (eme_trash_rsvp_action). 
And the function eme_trash_booking gets called from eme_trash_events ==> so when an event gets moved to trash (not permanently deleted), as you can see in the changeset. Make sure your hook is active in the backend of course. How do you define your hoook?

**Reply by f14m07** [reply]
My hook is "everywhere". But I had modify my hook some times ago and I think the pb is with my code. I will confirm

**Reply by f14m07** [reply]
Ok. The pb was me, I had changed my snippet on the real site but not on the local site. So now all is fine. Thanks for the code in eme_events.php

**Reply by liedekef** [reply]
Good to hear it works now.
Btw, I updated the EME doc on hooks a bit, look at the top of https://www.e-dynamics.be/wordpress/category/documentation/12-hooks-and-filters/ ==> EME supports uploaded snippets in a specific folder (so as to avoid yet another plugin, but maybe you need the snippets plugin for other modules of course)

**Reply by f14m07** [reply]
I use code snippet, but you prefer the first option, why? Is it quicker?

**Reply by liedekef** [reply]
less plugins, more security ... but there's no harm in using "code snippets" of course
